### DISCUSSION TITLE
Binning Folded Light curves

### DISCUSSION THREAD

**Discussion by rfreire-400** [lightkurve/lightkurve#1431] [Q&A] [answered]
Hello - I've been experimenting with the bin() and fold() methods. I'm able to bin and then fold, but it makes the eclipses barely visible due to the significant difference in data points. I was wondering if it's at all possible to do the opposite, fold and then bin, so I would get more data points at the eclipses. 

I've tried doing it, but it doesn't seem to be implemented. Is this a feature that could be added in a future update?
<img width="909" alt="Screenshot 2024-05-15 at 10 33 26 AM" src="https://github.com/lightkurve/lightkurve/assets/113457579/2f0d56d4-7403-4354-8b0f-8789b41bb643">

**Comment by orionlee** [comment, marked as answer]
binning folded lightcurve does make sense.

Unfortunately, there is an existing bug on binning folded lightcurve using `bins` parameter. See issue https://github.com/lightkurve/lightkurve/issues/1271 . 

Alternatives: 

1. parameter `n_bins` should work. See the [`bin()` documentation](https://docs.lightkurve.org/reference/api/lightkurve.LightCurve.bin.html) for the diffeerence.

2. One can also bin using `time_bin_size` parameter. E.g., the following will bin the folded lightcurve, with each bin has the duration of 10 minutes in the folded time.
```python
from astropy import units as u

lc_b = lc.fold(period=pg.period_at_max_power).bin(time_bin_size=10*u.miin)
```

3. If it's important to you to have a fixed number of bins, the following is anotherway to approximate it for 500 bins.
```python
lc_f = lc.fold(period=pg.period_at_max_power)
lc_binsize = (lc_f.time.max() - lc_f.time.min()).value / 500
lc_b = lc_f.bin(time_bin_size=lc_binsize)
```
The binned lightcurve above will have approximately 500 bins, if there is no significant data gap in the data, which is less likely given it's a folded lightcurve.

**Reply by rfreire-400** [reply]
Awesome, thanks a lot!
